src/strawberry_sqlalchemy_mapper/relay.py

```diff
@@ -0,0 +1,427 @@
+                    ),
+                    per_page=per_page,
+                )
+            )
+
+        if isinstance(session, AsyncSession):
+
+            async def resolve_async(nodes=nodes):
+                return await session.run_sync(lambda s: resolve_nodes(s))
+
+            return resolve_async()
+
+        return resolve_nodes(session)
+
+
+@overload
+def resolve_model_nodes(
+    source: Union[
+        Type[WithStrawberrySQLAlchemyObjectDefinition],
+        Type[relay.Node],
+        Type[_T],
+    ],
+    *,
+    session: Session,
+    info: Optional[Info] = None,
+    node_ids: Iterable[Union[str, relay.GlobalID]],
+    required: Literal[True],
+) -> AwaitableOrValue[Iterable[_T]]:
+    ...
+
+
+@overload
+def resolve_model_nodes(
+    source: Union[
+        Type[WithStrawberrySQLAlchemyObjectDefinition],
+        Type[relay.Node],
+        Type[_T],
+    ],
+    *,
+    session: Session,
+    info: Optional[Info] = None,
+    node_ids: None = None,
+    required: Literal[True],
+) -> AwaitableOrValue[Iterable[_T]]:
+    ...
+
+
+@overload
+def resolve_model_nodes(
+    source: Union[
+        Type[WithStrawberrySQLAlchemyObjectDefinition],
+        Type[relay.Node],
+        Type[_T],
+    ],
+    *,
+    session: Session,
+    info: Optional[Info] = None,
+    node_ids: Iterable[Union[str, relay.GlobalID]],
+    required: Literal[False],
+) -> AwaitableOrValue[Iterable[Optional[_T]]]:
+    ...
+
+
+@overload
+def resolve_model_nodes(
+    source: Union[
+        Type[WithStrawberrySQLAlchemyObjectDefinition],
+        Type[relay.Node],
+        Type[_T],
+    ],
+    *,
+    session: Session,
+    info: Optional[Info] = None,
+    node_ids: None = None,
+    required: Literal[False],
+) -> AwaitableOrValue[Optional[Iterable[_T]]]:
+    ...
+
+
+@overload
+def resolve_model_nodes(
+    source: Union[
+        Type[WithStrawberrySQLAlchemyObjectDefinition],
+        Type[relay.Node],
+        Type[_T],
+    ],
+    *,
+    session: Session,
+    info: Optional[Info] = None,
+    node_ids: Optional[Iterable[Union[str, relay.GlobalID]]] = None,
+    required: bool = False,
+) -> AwaitableOrValue[
+    Union[
+        Iterable[_T],
+        Query[_T],
+        Iterable[Optional[_T]],
+        Optional[Query[_T]],
+    ]
+]:
+    ...
+
+
+def resolve_model_nodes(
+    source,
+    *,
+    session: Session,
+    info=None,
+    node_ids=None,
+    required=False,
+    filter_perms=False,
+) -> AwaitableOrValue[
+    Union[
+        Iterable[_T],
+        Iterable[Optional[_T]],
+        Optional[Iterable[Optional[_T]]],
+    ]
+]:
+    """
+    Resolve model nodes.
+
+    Args:
+    ----
+        source:
+            The source model or the model type that implements the `Node` interface
+        info:
+            Optional gql execution info. Make sure to always provide this or
+            otherwise, the queryset cannot be optimized in case DjangoOptimizerExtension
```

Comment:
DjangoOptimizer mention is probably an artefact from the django implementation, right?